Changelog March 14 2024 (#256)
CloudCannon/platform-documentation: changelogs/2024-03-14_new-keys-to-allow-and-remove-custom-markup.mdx

---
_schema: default
title: New keys to allow and remove custom markup
type: minor
---

> This release added the `allow_custom_markup` key to enable editing of custom markup in rich text editors and the `remove_custom_markup` key to strip custom markup from rich text. It also added documentation links to checkboxes and switches, a&nbsp;*Copy output*&nbsp;button, custom URL support for your dashboard preview screenshot, nested keys inside of \`preview, and updated the&nbsp;*View output*&nbsp;button.

Reviewer: ```suggestion
This release added the `allow_custom_markup` key to enable editing of custom markup in rich text editors and the `remove_custom_markup` key to strip custom markup from rich text. It also added documentation links to checkboxes and switches, a *Copy output* button, custom URL support for your dashboard preview screenshot, nested keys inside of `preview`, and updated the *View output* button.
```

  * The `remove_custom_markup` key will strip custom markup from content edited in a rich text editor, allowing you to clean files of unnecessary HTML content.
  * The `remove_custom_markup` key is disabled if `allow_custom_markup` is enabled.
  * You can define these keys anywhere you would configure a rich text toolbar. By default, `allow_custom_markup` is `false` for inputs and `true` for editable regions, and `remove_custom_markup` is `false` for inputs and editable regions.
  * You can configure both keys using the configuration GUI.
  * When using `allow_custom_markup` outside of editable regions, content that uses JavaScript will not be editable for security reasons. These inputs will appear as uneditable snippets.
  * For more information, please read our documentation on best practices for HTML in [rich text](/documentation/articles/best-practices-for-rich-text/) and [editable regions](/documentation/articles/define-editable-regions-in-your-html/).
* You can now add documentation links to checkboxes and switches.
* You can now copy your build log using the *Copy output* button in the build modal, on the *Status* page, and in the *Builds* history tab.

> * You can now specify a custom URL for your dashboard preview screenshot using `editor.default_path`.

Reviewer: ```suggestion
* You can now specify a custom URL for your dashboard preview screenshot using `editor.default_path` in your global configuration file.
```

> * You can now specify a custom URL for your dashboard preview screenshot using `editor.default_path`.

Reviewer 2: Is it worth mentioning that this goes your CloudCannon config file?

* You can now use nested keys for any key inside `preview`.
* The&nbsp;*View output*&nbsp;button is now available while your build is optimizing assets.
* The embeds tool in the&nbsp;*WYSIWYG toolbar*&nbsp;now opens a code editor instead of a text area.
* Invalid editable regions will now preserve their original content and have better error messaging.

### Fixes:

* Fixed an issue where cloning or renaming a file from within an editing interface would not immediately open the cloned or renamed file.
* Fixed an issue where, on a Jekyll site, publishing a file to posts would not prompt you to add a date to the filename.
* Fixed an issue where, in the Visual Editor, updating the details for an image would cause the image panel to close.
* Fixed an issue where the&nbsp;*Context menu*&nbsp;for array and object inputs would show "Add undefined" instead of "Add item" for structures with no label configured.

> * Fixed a spacing issue affecting collapsed card lists.

Reviewer 2: I think "cards" are a concept mentioned in the docs, but maybe not "collapsed"? It might be more comprehensible to just say "arrays in the Data Editor".

> * You can now use nested keys for any key inside `preview`.
> * The&nbsp;*View output*&nbsp;button is now available while your build is optimizing assets.
> * The embeds tool in the&nbsp;*WYSIWYG toolbar*&nbsp;now opens a code editor instead of a text area.
> * Invalid editable regions will now preserve their original content and have better error messaging.
> 
> ### Fixes:
> 
> * Fixed an issue where cloning or renaming a file from within an editing interface would not immediately open the cloned or renamed file.
> * Fixed an issue where, on a Jekyll site, publishing a file to posts would not prompt you to add a date to the filename.
> * Fixed an issue where, in the Visual Editor, updating the details for an image would cause the image panel to close.
> * Fixed an issue where the&nbsp;*Context menu*&nbsp;for array and object inputs would show "Add undefined" instead of "Add item" for structures with no label configured.
> * Fixed a spacing issue affecting collapsed card lists.

Reviewer: ```suggestion
* You can now use nested keys for any key inside `preview`.
* The *View output* button is now available while your build is optimizing assets.
* The embeds tool in the *WYSIWYG toolbar* now opens a code editor instead of a text area.
* Invalid editable regions will now preserve their original content and have better error messaging.

### Fixes:

* Fixed an issue where cloning or renaming a file from within an editing interface would not immediately open the cloned or renamed file.
* Fixed an issue where, on a Jekyll site, publishing a file to posts would not prompt you to add a date to the filename.
* Fixed an issue where, in the Visual Editor, updating the details for an image would cause the image panel to close.
* Fixed an issue where the *Context menu* for array and object inputs would show "Add undefined" instead of "Add item" for structures with no label configured.
* Fixed a spacing issue affecting arrays in the Data Editor.
```

* Fixed an issue where the Data Editor would return to the top of the file when you changed data level.
* Fixed an issue where, for image inputs with resize configured, CloudCannon would unnecessarily duplicate images that already matched the specified height and width.
* Fixed an issue where, on branches that had a file with the same name as the branch, git pulls would not work.
* Fixed an issue preventing you from pushing local changes to a new branch.
* Fixed an issue where opening a file in an editing interface would automatically add focus to the first input.
* Fixed an issue where opening a file in an editing interface would trigger a change to that file without you updating any content.
* Fixed an issue where discarding your unsaved changes would not dismiss the conflicting updates warning.
* Fixed an issue where switching between editing interfaces on new files would create a second new file.
* Fixed an issue that sometimes prevented editable regions from loading for new files in the Visual Editor.
* Fixed an issue where embeds would delete their content.
* Improved output clarity by removing `data-cms-embed`.

> * Updated the color of inconsistent icons in the&nbsp;*WYSIWYG toolbar.*

Reviewer 2: ```suggestion
* Added consistency to the color of icons in the *WYSIWYG toolbar.*
```

> * Updated the copy in the&nbsp;*Switch to editing*&nbsp;banner and the Legacy&nbsp;*Editing locked*&nbsp;banner.

Reviewer: ```suggestion
* Updated the copy in the *Switch to editing* banner and the Legacy *Editing locked* banner.
```
